File: Modules/SekaiMasterUpdater/compare.py
def compare_version(new_version, current_version) -> bool:
    new_version_parts = new_version.split(".")
    current_version_parts = current_version.split(".")

    for i in range(len(new_version_parts)):
        if int(new_version_parts[i]) > int(current_version_parts[i]):
            return True
        elif int(new_version_parts[i]) < int(current_version_parts[i]):
            return False

    return False

File: updater/compare.go
package updater

import (
	"github.com/hashicorp/go-version"
)

func CompareVersion(newVersion, currentVersion string) (bool, error) {
	v1, err := version.NewVersion(newVersion)
	if err != nil {
		return false, err
	}
	v2, err := version.NewVersion(currentVersion)
	if err != nil {
		return false, err
	}
	return v1.GreaterThan(v2), nil
}
